### ISSUE TITLE
Updating annotations inside loop without freezing page.annots() as list causes infinite loop or segfault

### ISSUE THREAD

**Issue by matoles** [pymupdf/PyMuPDF#1588]
_**Please provide all mandatory information!**_

## Describe the bug (mandatory)
The following code does not terminate, or under certain PDF conditions crashes without error code. Observing the console output during non-termination suggests infinitely many new annotations are being created, possibly as a result of some race condition. The behavior can be mitigated by iterating over `list(page.annots())` rather than `page.annots()` but this workaround does not appear documented.

Source PDF requires a minimum number of pages and annotations to exhibit bug behavior. My PDF contains approximately 40 annotations on each of 5 out of 12 pages (~200 total)

## To Reproduce (mandatory)
```
import fitz


def review_annotations(
    pdf,
):
    with fitz.open(stream=pdf, filetype="pdf") as doc:
        for i, page in enumerate(doc):
            # for j, annot in enumerate(list(page.annots())): # works
            for j, annot in enumerate(page.annots()): # doesn't work
                annot.set_colors(
                    {
                        "fill": [1.0, 0.0, 0.0],
                    }
                )
                annot.set_opacity(0.5)
                annot.set_blendmode("dodge")
                annot.update()
                print(i, j)
        print("done")
        return doc.save("output.pdf")


pdf_path = (
    "path_to_file.pdf"
)

with open(pdf_path, "rb") as f:
    review_annotations(f.read())

```

## Your configuration (mandatory)
```
print(sys.version, "\n", sys.platform, "\n", fitz.__doc__)
3.8.5 (default, Sep  4 2020, 02:22:02) 
[Clang 10.0.0 ] 
 darwin 
 
PyMuPDF 1.19.1: Python bindings for the MuPDF 1.19.0 library.
Version date: 2021-10-23 00:00:01.
Built for Python 3.8 on darwin (64-bit).
```
 - PyMuPDF installed via pip.

**Comment by JorjMcKie** [comment]
To be acceptable as a bug, a reproducing file is required.

**Comment by JorjMcKie** [comment]
I understand that you will not share data for reproducing the error.
I am therefore going to close this issue and its twin #1587.
But let me give you a few hints for how you can still implement your app's intention.

Background:
1. Because of significant feature extensions of PyMuPDF (vs "native" MuPDF), annotation updates in many cases require reloading the page to become persistent. This requirement has aggravated in the recent MuPDF version 1.19.0. To be sure, **_page reloading should always be done_** after an annotation creation or update.
2. Reloading a page entails creating **_a new page object_** (and dropping the old one). This fact prohibits the use of method `Document.reload_page()` within the scope of generator `Page.annots()`. The reasons for this should be obvious.

So bottom line of the above is: you **_cannot update annotations within the scope_** of a `Page.annots()`!

To still achieve your goal, use the following coding scheme:

```python
xrefs = [annot.xref for annot in page.annots()]
for xref in xrefs:
    annot = page.load_annot(xref)
    # ... update desired annot properties
    annot.update()
    page = doc.load_page(page)
```
Please especially note the need for using a list of cross reference numbers here: A list of annotations would not work, because the original page's object id is part of every annotation's properties.

The next PyMuPDF version will be immunized against page reloading within the annotation iterator.

**Comment by JorjMcKie** [comment]
Closed for lack of reproducible data.

**Comment by matoles** [comment]
Sorry about the delay. Here is the input and output files. The output should express the move/color-change behavior.

[calibration.pdf](https://github.com/pymupdf/PyMuPDF/files/8062650/calibration.pdf)
[output.pdf](https://github.com/pymupdf/PyMuPDF/files/8062654/output.pdf)
